pkg/translator/prometheusremotewrite/helper.go

```diff
@@ -340,12 +340,13 @@ func addSingleHistogramDataPoint(pt pmetric.HistogramDataPoint, resource pcommon
 	infBucket := &prompb.Sample{
 		Timestamp: time,
 	}
-	if pt.Flags().HasFlag(pmetric.MetricDataPointFlagNoRecordedValue) {
+	if pt.Flags().HasFlag(pmetric.MetricDataPointFlagNoRecordedValue) || pt.BucketCounts().Len() < 1 {
```

Comment:
```suggestion
	if pt.Flags().HasFlag(pmetric.MetricDataPointFlagNoRecordedValue) || pt.BucketCounts().Len() == 0 {
```

The length of a slice can't be less than zero, so this is more representative of intent.  This makes clearer that this is a case of bad input that should probably be validated earlier and never make it this far.  The metric proto requires that there be [at least one entry](https://github.com/open-telemetry/opentelemetry-proto/blob/main/opentelemetry/proto/metrics/v1/metrics.proto#L424-L426) in the bucket count list:

>   // The number of elements in bucket_counts array must be by one greater than
>  // the number of elements in explicit_bounds array.
>  repeated fixed64 bucket_counts = 6;

Even if the explicit bounds are empty then there must still be an entry in the `+Inf` bucket.

Reply:
In this scenario where something prior is failing to adhere to this, what would be the preferred scenario?

- In the event that a histogram metric has no values, then append an inf+ bucket with the value zero?
- In the event that a histogram metric has no values, ignore the metric? 

I think in terms of ensuring that data is correct here that this approach is the safest ensuring that data is correctly processed but I am not sure on the implications.

Reply:
I update to == 0 and fixed the check to check for value.IsStaleNaN. I am not sure what the implication would be, but I would like to get the Go Panic fixed first.

Reply:
I think what this PR is doing is probably wrong.  It is conflating the Prometheus use of a signaling `NaN` to convey staleness information with invalid histogram data points.  I suspect that the data point here should simply be ignored as early as possible.

Reply:
The purpose of this PR is to stop the entire collector from crashing because of bad input. I made the PR as simple as possible in order to not affect any cases except a panic. This basically follows the same pattern as when the metric indicates it has no data "pmetric.MetricDataPointFlagNoRecordedValue". I understand that this data should probably be thrown out earlier, however that introduced possible additional behavior changes. With the lack of test coverage I felt this would have the least risk of introducing more problems because it will only affect data which would have resulted in a crash. 

If this is not an appropriate step forward I can close the PR and delete my fork.

Reply:
> If this is not an appropriate step forward I can close the PR and delete my fork.

You'll need to forgive us for being overly critical, the collector is depended on by a lot of users now so ensuring correctness of implementation is important. Don't let this be a discouragement to contribute but rather we really want you fix to work for all possible use cases / deployments. <3 

@Aneurysm9 , would you feel more comfortable with this change going in as is if there was a follow up issue to address the actual cause?